client/components/Treechart/index.tsx

```diff
@@ -0,0 +1,249 @@
+  CoinRateType,
+  CoinRateContentType
+} from '@/types/ChartTypes'
+
+const coinIntervalRate = 1000
+
+const updateChart = (
+  svgRef: React.RefObject<SVGSVGElement>,
+  data: CoinRateContentType[],
+  width: number,
+  height: number
+) => {
+  if (!svgRef.current) return
+  const chartContainer = d3.select<SVGSVGElement, CoinRateContentType>(
+    svgRef.current
+  )
+  chartContainer.attr('width', width)
+  chartContainer.attr('height', height)
+  const chartArea = d3.select('svg#chart-area')
+  const [min, max]: number[] = [
+    d3.min(data, d => Math.abs(d.value)) as number,
+    d3.max(data, d => d.value) as number
+  ]
+  const treeMapvalueScale = d3
+    .scaleLinear()
+    .domain([min, max])
+    .range([0.5, 1.5])
+  const root: d3.HierarchyNode<CoinRateContentType> = d3
+    .stratify<CoinRateContentType>()
+    .id(function (d): string {
+      return d.name
+    })
+    .parentId(function (d): string {
+      return d.parent
+    })(data)
+  const sort = (
+    a: d3.HierarchyNode<CoinRateContentType>,
+    b: d3.HierarchyNode<CoinRateContentType>
+  ) => d3.descending(a.value, b.value)
+  //const sort = (a, b) => d3.ascending(a.id, b.id)
+
+  root
+    .sum(function (d): number {
+      return Math.abs(+d.value)
+    })
+    .sort(sort)
+
+  d3.treemap<CoinRateContentType>().size([width, height]).padding(4)(root)
+
+  chartArea
+    .selectAll<SVGSVGElement, CoinRateContentType>('g')
+    .data<d3.HierarchyRectangularNode<CoinRateContentType>>(
+      root.leaves() as Array<d3.HierarchyRectangularNode<CoinRateContentType>>
+    )
+    .join(
+      enter => {
+        const $g = enter.append('g')
+        $g.append('rect')
+          .attr('x', function (d) {
+            return d.x0
+          })
+          .attr('y', function (d) {
+            return d.y0
+          })
+          .attr('width', function (d) {
+            return d.x1 - d.x0
+          })
+          .attr('height', function (d) {
+            return d.y1 - d.y0
+          })
+          .attr('fill', function (d) {
+            return d.data.value > 0 ? 'red' : 'blue'
+          })
+          .attr('opacity', function (d) {
+            return treeMapvalueScale(Math.abs(d.data.value as number))
+          })
+          .style('stroke', 'black')
+        $g.append('text')
+          .attr('x', function (d) {
+            return d.x0 + Math.abs(d.x1 - d.x0) / 2
+          })
+          .attr('y', function (d) {
+            return d.y0 + Math.abs(d.y1 - d.y0) / 2
+          })
+          .attr('text-anchor', 'middle')
+          .text(function (d) {
+            return (
+              d.data.name + '\n' + String(Number(d.data.value).toFixed(2)) + '%'
+            )
+          })
+          .style('font-size', function (d) {
+            return `${(d.x1 - d.x0) / 9}px`
+          })
+          .attr('fill', 'white')
+        return $g
+      },
+      update => {
+        update
+          .select('rect')
+          .transition()
+          .duration(500)
```

Comment:
아까 확인했을 때는 transition이 잘 적용되는 것으로 확인했었습니다만, 이런 내용이 있어서 한 번만 확인 부탁드립니다 !

https://www.d3indepth.com/transitions/
Updating elements
You can apply a transition to entering and updating elements by adding a call to .transition after the .join call. This will cause all elements (except exiting elements) to transition.

ex)
d3.select('svg')
  .selectAll('circle')
  .data(data)
  .join(
    function(enter) { ... },
    function(update) { ... },
    function(exit) { ... }
  )
  .transition()
  .attr('cx', function(d) {
    return d;
  })
  .style('opacity', 0.75);

Reply:
이런식으로 하면 모든 행위에 대해 transition을 일괄적용할 수 있군요. 감사합니다.